wireup/ioc/validation.py

```diff
@@ -94,6 +95,37 @@ def assert_dependency_exists(container: BaseContainer, parameter: AnnotatedParam
         raise WireupError(msg)
 
 
+def assert_valid_resolution_path(
+    container: BaseContainer, dependency: AnnotatedParameter, path: list[tuple[AnnotatedParameter, Any]]
+) -> None:
+    """Assert that the resolution path for a dependency does not create a cycle."""
+    if dependency.klass in container._registry.interfaces or dependency.is_parameter:
+        return
+    dependency_service_factory = container._registry.factories.get((dependency.klass, dependency.qualifier_value))
```

Comment:
Here we can actually just access the dict directly since all dependencies have a factory since wireup needs to know how to create them.

`dependency_service_factory = container._registry.factories[dependency.klass, dependency.qualifier_value]`